### PR TITLE
Feature/internal interfaces

### LINKED ISSUE #56: Extract interfaces+mocks into internal package

### PR DESCRIPTION
**Pull request by jelemux**
Resolves #56

### CODE REVIEW

File: controllers/doguUpgradeManager.go [hunk 1]
```diff
@@ -3,22 +3,21 @@ package controllers
...
 	"github.com/cloudogu/k8s-dogu-operator/controllers/cesregistry"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/config"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/dependency"
+	"github.com/cloudogu/k8s-dogu-operator/controllers/exec"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/health"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/imageregistry"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/resource"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/serviceaccount"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/upgrade"
+	internal2 "github.com/cloudogu/k8s-dogu-operator/internal"
```
Comment (ppxl) [inline controllers/doguUpgradeManager.go:20]: why is this import named in this way? Was there a "first" internal package in the first place?

File: internal/exec.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// PodStatus describes a state in the lifecycle of a pod.
+type PodStatus string
+
+const (
+	// ContainersStarted means that all containers of a pod were started.
+	ContainersStarted PodStatus = "started"
+	// PodReady means that the readiness probe of the pod has succeeded.
+	PodReady PodStatus = "ready"
+)
+
```
Comment (ppxl) [inline internal/exec.go:68]: Collecting interfaces in an internal package is one thing, I fully understand the reason. But here we have actual values which from outside are referenced as `internal.PodReady`. The package helps to identify the business value of a thing. And now, by moving here, the business context is gone.

If we are not careful we might encourage other developers to add further values here, also void of business context. So how do we solve this situation?

File: internal/mocks/external/external.go [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Package external implements mocks that implement 3rd party interfaces, t. i. interfaces which we do not control.
```
Comment (ppxl) [inline internal/mocks/external/external.go:1]: I took the freedom and added a package comment in order to explain the `internal/.../external` situation.

File: retry/retry.go [hunk 4]
```diff
@@ -29,10 +29,10 @@ func (tre *TestableRetrierError) Error() string {
 	return tre.Err.Error()
 }
 
-// OnErrorRetry provides a K8s-way "retrier" mechanism. The value from retriable is used to indicate if workload should
+// OnError provides a K8s-way "retrier" mechanism. The value from retriable is used to indicate if workload should
 // retried another time. Please see AlwaysRetryFunc() if a workload should always retried until a fixed threshold is
 // reached.
-func OnErrorRetry(maxTries int, retriable func(error) bool, workload func() error) error {
+func OnError(maxTries int, retriable func(error) bool, workload func() error) error {
```
Comment (ppxl) [inline retry/retry.go:35]: Ah, I see, you got rid of repeating the package name. Good job!

**Review by ppxl — CHANGES_REQUESTED** [review]
Good job, and I am inclined to merge wasn't it for the one thing of constants in `internal/exec.go`.  We should discuss this matter

File: internal/mocks/external/external.go (hunk 3 above)
Comment (jelemux) [reply internal/mocks/external/external.go:1]: Thanks!

**Review by ppxl — APPROVED** [review]
LGTM

**Pull request merged**